Firestore/Source/API/FSTUserDataConverter.mm

```diff
@@ -62,7 +63,7 @@ - (instancetype)initWithData:(FSTObjectValue *)data
   return self;
 }
 
-- (NSArray<FSTMutation *> *)mutationsWithKey:(FSTDocumentKey *)key
+- (NSArray<FSTMutation *> *)mutationsWithKey:(const DocumentKey &)key
```

Comment:
Nit: in `FIRDocumentSnapshot.mm`, `key` is taken by value both in the class method and the instance method, and moved twice. Here, it is taken by const reference in the class method, but by value in object method. Is there a reason for this difference?

Reply:
Only constructor a.k.a. initializer are taking value and moves the key. Non-constructor method can take value but seems adding complexity to the logic. For example, the code here is of this structure:

   call bar with key;
   if (cond) call foo with key;

We can only make the last call via move; however that call is under a condition... code changes may eventually break it in a way such that key is moved but later used.